### PR TITLE
fix(update dependencies): Update dependencies and adjust code and settings

### LINKED ISSUE #251: Update dependencies
Update dependencies to fix vulnerabilities

### PR DESCRIPTION
**Pull request by Retro64**
close #251

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ const extractAccessToken = (authHeader: string): string | undefined => {
 const setTokeninfo = (req: Request): (data: Token) => void => {
   return (data: Token) => {
 
-    const tokeninfo = { ...data };
+    const tokeninfo = { ...data, ...{access_token: undefined} };
```
Comment (Retro64) [inline src/utils.ts:114]: Note: TS 4.x.x enforces the property to be deleted needs to be an optional - I guess this is less invasive than making the property optional in the type

**Comment by Retro64** [comment]
:+1:

File: src/utils.ts (hunk 1 above)
Comment (bzums) [reply src/utils.ts:114]: `const tokeninfo = { ...data, access_token: undefined };`

File: package.json [hunk 2]
```diff
@@ -25,39 +25,39 @@
...
-    "@typescript-eslint/eslint-plugin-tslint": "3.2.0",
-    "@typescript-eslint/parser": "3.2.0",
+    "@sinonjs/fake-timers": "7.0.2",
+    "@types/chai": "4.2.14",
+    "@types/chai-as-promised": "7.1.3",
+    "@types/express": "4.17.11",
+    "@types/mocha": "8.2.0",
+    "@types/node-fetch": "2.5.8",
+    "@types/sinon-chai": "3.2.5",
+    "@types/sinonjs__fake-timers": "6.0.2",
```
Comment (ISO50) [inline package.json:41]: What the difference to `"@sinonjs/fake-timers": "7.0.2",`?

File: src/utils.ts (hunk 1 above)
Comment (ISO50) [reply src/utils.ts:114]:
```suggestion
    const tokeninfo = { 
       ...data, 
       access_token: undefined
    };
```

Comment (ISO50) [reply src/utils.ts:114]: Same idea, oneline or multiline doesn't matter for me, but destructuring shouldn't be needed.

File: package.json (hunk 2 above)
Comment (bzums) [reply package.json:41]: I just learned this when having a look at koa...this is the way you name `@types` packages which are already scoped (`@sinonjs` in this case - yeah, that is not nice...)

Comment (Retro64) [reply package.json:41]: These are the types - sinon is currently implementing types on their own, but are not completely done yet - see https://github.com/sinonjs/fake-timers#help-us-get-our-typescript-definitions-production-ready

Comment (Retro64) [reply package.json:41]: Therefore also the distinct commit

**Comment by Retro64** [comment]
:+1:

**Comment by ISO50** [comment]
👍

**Pull request merged**